src/prophecy/layer/dense_layer.hh

```diff
@@ -47,7 +53,16 @@ void DenseLayer<T>::compile(std::weak_ptr<Layer<T>>   prev,
     // Initialize weights and delta_weights
     const std::vector<uint> w_shape = {this->nb_neurons_,
                                        prev.lock()->get_out_shape()[0]};
-    this->weights_                  = xt::random::randn<T>(w_shape, 0, 1);
+    if (init_ == "xavier")
+    {
+        const T weights_std = sqrt(((T)2) / (w_shape.front(), w_shape.back()));
```

Comment:
- Shouldn't 2 be divided by `(w_shape.front() + w_shape.back())`?
- To make the code cleaner it would be great to `static_cast<T>()` the result instead of using `(T)2`.